src/main/scala/com/microsoft/hyperspace/actions/RefreshIncrementalAction.scala

```diff
@@ -0,0 +1,124 @@
+ * - Create new index version on these files.
+ * - Update metadata to reflect the latest snapshot of index. This snapshot includes all the old
+ *   and the newly created index files. The source content points to the latest data files.
+ *
+ * @param spark SparkSession.
+ * @param logManager Index LogManager for index being refreshed.
+ * @param dataManager Index DataManager for index being refreshed.
+ */
+class RefreshIncrementalAction(
+    spark: SparkSession,
+    logManager: IndexLogManager,
+    dataManager: IndexDataManager)
+    extends RefreshActionBase(spark, logManager, dataManager) {
+  final override def op(): Unit = {
+    logInfo(
+      "Refresh index is updating index by removing index entries " +
+        s"corresponding to ${deletedFiles.length} deleted source data files.")
+
+    if (appendedFiles.nonEmpty) {
+      val dfWithAppendedFiles = {
+        val relation = previousIndexLogEntry.relations.head
+        // Create a df with only diff files from original list of files.
+        spark.read
+          .schema(df.schema)
+          .format(relation.fileFormat)
+          .options(relation.options)
+          .load(appendedFiles.map(_.name): _*)
+      }
+      write(spark, dfWithAppendedFiles, indexConfig)
+    }
+
+    if (deletedFiles.nonEmpty) {
+      val refreshDF =
+        spark.read
+          .parquet(previousIndexLogEntry.content.files.map(_.toString): _*)
+          .filter(
+            !col(s"${IndexConstants.DATA_FILE_NAME_COLUMN}").isin(deletedFiles.map(_.name): _*))
+
+      refreshDF.write.saveWithBuckets(
+        refreshDF,
+        indexDataPath.toString,
+        previousIndexLogEntry.numBuckets,
+        indexConfig.indexedColumns,
+        SaveMode.Append)
+    }
+  }
+
+  /**
+   * Validate index is in active state for refreshing and there are some changes
+   * in source data file(s).
+   */
+  final override def validate(): Unit = {
+    super.validate()
+
+    if (appendedFiles.isEmpty && deletedFiles.isEmpty) {
+      throw NoChangesException("Refresh incremental aborted as no source data change found.")
+    }
+
+    if (deletedFiles.nonEmpty && !previousIndexLogEntry.hasLineageColumn(spark)) {
+      throw HyperspaceException(
+        "Index refresh (to handle deleted source data) is " +
+          "only supported on an index with lineage.")
+    }
+  }
+
+
+  /**
+   * Create a log entry with all data files, and index content merged with previous index content.
+   * This contains ALL data files (files which were indexed previously, and files which are being
+   * indexed in this operation). It also contains ALL index files (index files for previously
+   * indexed data as well as newly created files).
+   *
+   * @return Merged index log entry.
+   */
+  override def logEntry: LogEntry = {
+    // Log entry with complete data and newly index files.
+    val entry = getIndexLogEntry(spark, df, indexConfig, indexDataPath)
+
+    if (deletedFiles.isEmpty) {
```

Comment:
It would be helpful if you add a comment, explaining in more details, why having deletedFiles affects the way new entry is created.